### PR TITLE
Add retrofit docs and sample

### LINKED ISSUE #88: Add retrofit integration docs and sample
We need to add the Helios retrofit integration to the microsite documentation and also add a little example on the `samples` module (or in a new module under the same folder)

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #88
This PR comes from #95 with some improvements

### CODE REVIEW

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:warning: `Decoding` (Threshold: 262.60789022869585)

Benchmark | Value
|--|--|
master_benchmark | 11291.570280322514
helios_benchmark | 11168.267484478209


:heavy_check_mark: `Decodingfromraw` (Threshold: 250.0)

Benchmark | Value
|--|--|
master_benchmark | 7083.169867857131
helios_benchmark | 7207.277756728222


:red_circle: `Parsing` (Threshold: 250.0)

Benchmark | Value
|--|--|
master_benchmark | 22096.89998013556
helios_benchmark | 21140.734722670804

File: helios-samples/retrofit-sample/src/main/kotlin/helios/sample/retrofit/Sample.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        )
+      )
+      .baseUrl("https://httpbin.org/")
+      .build()
+
+    val api = retrofit.create(HttpBinApi::class.java)
+
+    val response = api.getJson().execute()
+
+    print(response.body())
```
Comment (fedefernandez) [inline helios-samples/retrofit-sample/src/main/kotlin/helios/sample/retrofit/Sample.kt]: Do we have a way of using some log library or at least putting that in an effect?

File: helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt [hunk 2]
```diff
@@ -1,24 +1,21 @@
...
-      .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
-      .flatMap { it.decode(decoder) }
-      .getOrElse { null }
-  } catch (ex: Exception) {
-    null
-  } finally {
-    value.close()
+      .parseFromByteBuffer(ByteBuffer.wrap(body.byteStream().readBytes()))
+      .flatMap { json -> json.decode(decoder).mapLeft { IOException("Decode failed with the error: $it") } }
+      .fold({ throw IOException(it) }, { it })
```
Comment (fedefernandez) [inline helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt]: This is creating an `IOException` inside of an `IOException`. Should this be something like:

```suggestion
      .fold({ throw it }, { it })
```

Do we have a `rethrow` like in cats?

**Comment by MaureenElsberry** [comment]
Good to merge @AdrianRaFo!

File: helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt (hunk 2 above)
Comment (nomisRev) [reply helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt]: @fedefernandez We do in the SNAPSHOT Arrow but not in stable.

Comment (AdrianRaFo) [reply helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt]: I'll add a TODO

**Comment by AdrianRaFo** [comment]
@fedefernandez comments addressed

**Pull request merged**